File: Lab1/circle/Parabola.py
def parabola_draw(event_1, event_2):

    x1, y1 = event_1.x, event_1.y
    x2, y2 = event_2.x, event_2.y

    print(x1, y1)
    print(x2, y2)

    a = (y2 - y1) / ((x2 - x1) ** 2)
    b = -2 * a * x1
    c = y1 - a * x1 ** 2 - b * x1

    x = min(x1, x2)
    x_max = max(x1, x2)
    y_values = []

    while x <= x_max:
        y = a * x ** 2 + b * x + c
        y_values.append(y)
        x += 1

    x_values = list(range(min(x1, x2), max(x1, x2) + 1))

    print(x_values)
    pixels = []

    if x1 < x2:
        t = 0
        for i in range(len(x_values)):
            pixels.append((x_values[i], y_values[i]))
            if i > 0:
                pixels.append((x_values[i] - abs(x_values[0] - x_values[i]) - t, y_values[i]))
            t += 1
    else:
        t = 0
        for i in range(len(x_values)):
            pixels.append((x_values[i], y_values[i]))
            if i > 0:
                pixels.append((x_values[len(x_values) - i] + 1 + abs(max(x_values) - min(x_values)), y_values[i]))
            t += 1


    print(pixels)
    return pixels

File: Lab1-2/Circle/Parabola.cs
using System.Windows;

namespace Lab1_2.Circle;

public class Parabola
{
    public static List<Point> Draw(Point event1, Point event2)
    {
        var x1 = event1.X;
        var y1 = event1.Y;
        var x2 = event2.X;
        var y2 = event2.Y;

        Console.WriteLine(x1 + " " + y1);
        Console.WriteLine(x2 + " " + y2);

        var a = (y2 - y1) / Math.Pow((x2 - x1), 2); 
        var b = -2 * a * x1;
        var c = y1 - a * Math.Pow(x1, 2) - b * x1;

        var x = Math.Min(x1, x2);
        var xMax = Math.Max(x1, x2);
        var yValues = new List<double>();

        while (x <= xMax)
        {
            var y = (a * Math.Pow(x, 2) + b * x + c);
            yValues.Add(y);
            x++;
        }

        var xValues = new List<double>();
        for (var i = Math.Min(x1, x2); i <= Math.Max(x1, x2); i++)
        {
            xValues.Add(i);
        }

        Console.WriteLine(string.Join(", ", xValues));
        var pixels = new List<Point>();

        if (x1 < x2)
        {
            var t = 0;
            for (var i = 0; i < xValues.Count; i++)
            {
                pixels.Add(new Point(xValues[i], yValues[i]));
                if (i > 0)
                {
                    pixels.Add(new Point(xValues[i] - Math.Abs(xValues[0] - xValues[i]) - t, yValues[i]));
                }
                t++;
            }
        }
        else
        {
            var t = 0;
            for (var i = 0; i < xValues.Count; i++)
            {
                pixels.Add(new Point(xValues[i], yValues[i]));
                if (i > 0)
                {
                    pixels.Add(new Point(xValues[^i] + 1 + Math.Abs(xValues[^1] - xValues[0]), yValues[i]));
                }
                t++;
            }
        }

        foreach (var pixel in pixels)
        {
            Console.WriteLine("(" + pixel.X + ", " + pixel.Y + ")");
        }

        return pixels;
    }
}